src/main/java/dev/openfga/sdk/api/client/OpenFgaClient.java

```diff
@@ -383,7 +387,29 @@ public CompletableFuture<ClientCheckResponse> check(ClientCheckRequest request,
      *
      * @throws FgaInvalidParameterException When the Store ID is null, empty, or whitespace
      */
-    // TODO
+    public CompletableFuture<List<ClientCheckResponse>> batchCheck(
+            List<ClientCheckRequest> requests, ClientBatchCheckOptions options) {
+        int maxParallelRequests = options.getMaxParallelRequests() != null
+                ? options.getMaxParallelRequests()
+                : DEFAULT_MAX_METHOD_PARALLEL_REQS;
+        var executor = Executors.newWorkStealingPool(maxParallelRequests);
+
+        var responses = new ConcurrentLinkedQueue<ClientCheckResponse>();
```

Comment:
TODO: When testing look into other concurrent collection types here.

Reply:
ConcurrentLinkedQueue is the thing to do. Non-blocking beyond CAS instructions and can't find a way to break it.